developer_manual/digging_deeper/flow.rst

```diff
@@ -2,8 +2,87 @@
+- ``\OCP\Files::postWrite``
+- ``\OCP\Files::postRename``
+- ``\OCP\Files::postDelete``
+- ``\OCP\Files::postTouch``
+- ``\OCP\Files::postCopy``
+
+Listeners
+=========
+
+Event listeners are classes registered with the manager, along with an array of events they should listen for.
+
+The first event listener your application should implement is for the ``RegisterOperationsEvent`` event.
+
+This event is dispatched when the workflow engine resolves which event listeners should receive a broadcasted event.
+
+This initial event listener should register your operations and the name of the JS bundle containing a Vue component for configuring conditions for your event listeners.
+
+`An example RegisterOperationsEvent listener can be found here <https://github.com/nextcloud/workflow_pdf_converter/blob/master/lib/Listener/RegisterFlowOperationsListener.php>`_.
+
+Register this event listener within the ``register`` function of your ``Application`` class::
+
+    public function register(IRegistrationContext $context): void {
+        $context->registerEventListener(
+            RegisterOperationsEvent::class, 
+            RegisterFlowOperationsListener::class
+        );
+    }
+
+Operations
+==========
+
+Operations are classes registered with one or more event listeners.
+
+This class must implement one of the following `IOperation <https://github.com/nextcloud/server/blob/master/lib/public/WorkflowEngine/IOperation.php#L33>`_ interfaces:
+
+- `ISpecificOperation <https://github.com/nextcloud/server/blob/master/lib/public/WorkflowEngine/ISpecificOperation.php>`_ – for operations designed to work with one entity type.
+- `IComplexOperation <https://github.com/nextcloud/server/blob/master/lib/public/WorkflowEngine/IComplexOperation.php>`_ – for operations that perform their own trigger logic.
+
+When an event is fired that your operation listens for, it will call the onEvent function of your operation.
+
+`An example ConvertPDFOperation can be found here <https://github.com/nextcloud/workflow_pdf_converter/tree/master/lib/Operation.php>`_.
+
+Register the operation against the event in the ``handle`` function of your ``RegisterOperationsEvent`` listener.
+
+Configuration Component
+=======================
+
+The configuration component is what users see when they add your flow and configure its rules.
+
+The ``RegisterOperationsEvent`` listener you created earlier registers the name of the JS bundle.
+
+This JS bundle should be compiled using ``@nextcloud/webpack-vue-config`` as follows:
+
+First, create a `webpack.js` file in the root of your application's folder::
+
+    const webpackConfig = require('@nextcloud/webpack-vue-config')
+
+    module.exports = webpackConfig
+
+Then, create ``src/main.js`` with the following contents::
+
+    import ConvertToPdf from './ConvertToPdf'
+
+    OCA.WorkflowEngine.registerOperator({
+        id: 'OCA\\WorkflowPDFConverter\\Operation',
+        operation: 'keep;preserve',
+        options: ConvertToPdf,
```

Comment:
This is "legacy". going forward element should be used and allows apps to provide their content isolated, so it does not cause conflicts when the app and server use different Vue versions, etc. See the following PR and from there linked PRs and issues:
https://github.com/nextcloud/spreed/pull/14367